app/models/start_teacher_training.rb

```diff
@@ -2,6 +2,14 @@ class StartTeacherTraining < Base
   attribute :year_of_entry, :string
 
   validates :year_of_entry, inclusion: { in: %w(2020 2021 2022) }
+  
+  def year_range(number_of_years) # sets year range for view
```

Comment:
Swap the `each` for a `map` and the `to_a` shouldn't be required (I think) then you can drop the `years` initialisation and return the `map` directly:

```
(Date.today.year..Date.today.next_year(number_of_years).year).map do |year|
  years << OpenStruct.new(value: year, name: year)
end
```

Reply:
Nice. Also on looking at it again had to change the validations, I think we may have to come back to it anyway as it may be covered by an API call?